src/devices/cpu/ht1130/ht1130.cpp

```diff
@@ -0,0 +1,1051 @@
+{
+}
+
+std::unique_ptr<util::disasm_interface> ht1130_device::create_disassembler()
+{
+	return std::make_unique<ht1130_disassembler>();
+}
+
+device_memory_interface::space_config_vector ht1130_device::memory_space_config() const
+{
+	return space_config_vector{
+		std::make_pair(AS_PROGRAM, &m_space_config),
+		std::make_pair(AS_DATA, &m_extregs_config)
+	};
+}
+
+void ht1130_device::internal_map(address_map &map)
+{
+	map(0x000, 0xfff).rom();
+}
+
+void ht1130_device::tempram_w(offs_t offset, u8 data)
+{
+	m_tempram[offset] = data & 0xf;
+}
+
+void ht1130_device::displayram_w(offs_t offset, u8 data)
+{
+	m_displayram[offset] = data & 0xf;
+
+	// there might be a better way to do this
+	m_display_offset_out(offset);
+	m_display_data_out(m_displayram[offset]);
+}
+
+void ht1130_device::setreg(u8 which, u8 data)
+{
+	m_regs[which] = data & 0xf;
+}
+
+u8 ht1130_device::getreg(u8 which)
+{
+	return m_regs[which] & 0xf;
+}
+
+void ht1130_device::setacc(u8 data)
+{
+	m_acc = data & 0xf;
+}
+
+u8 ht1130_device::getacc()
+{
+	return m_acc & 0xf;
+}
+
+u8 ht1130_device::getcarry()
+{
+	return m_carry & 1;
+}
+
+void ht1130_device::setcarry()
+{
+	m_carry = 1;
+}
+
+void ht1130_device::clearcarry()
+{
+	m_carry = 0;
+}
+
+void ht1130_device::settimer(u8 data)
+{
+	m_timer = data;
+}
+
+void ht1130_device::settimer_upper(u8 data)
+{
+	m_timer = (m_timer & 0xf) | (data & 0xf) << 4;
+}
+
+void ht1130_device::settimer_lower(u8 data)
+{
+	m_timer = (m_timer & 0xf0) | (data & 0xf);
+}
+
+u8 ht1130_device::gettimer_upper()
+{
+	return (m_timer >> 4) & 0xf;
+}
+
+u8 ht1130_device::gettimer_lower()
+{
+	return m_timer & 0xf;
+}
+
+u8 ht1130_device::getr1r0()
+{
+	return (getreg(1)<<4) | getreg(0);
+}
+
+u8 ht1130_device::getr1r0_data()
+{
+	const u8 dataaddress = getr1r0();
+	return space(AS_DATA).read_byte(dataaddress);
+}
+
+void ht1130_device::setr1r0_data(u8 data)
+{
+	const u8 dataaddress = getr1r0();
+	space(AS_DATA).write_byte(dataaddress, data);
+}
+
+u8 ht1130_device::getr3r2()
+{
+	return (getreg(3)<<4) | getreg(2);
+}
+
+u8 ht1130_device::getr3r2_data()
+{
+	const u8 dataaddress = getr3r2();
+	return space(AS_DATA).read_byte(dataaddress);
+}
+
+void ht1130_device::setr3r2_data(u8 data)
+{
+	const u8 dataaddress = getr3r2();
+	space(AS_DATA).write_byte(dataaddress, data);
+}
+
+
+void ht1130_device::internal_data_map(address_map &map)
+{
+	map(0x00, 0x7f).ram().w(FUNC(ht1130_device::tempram_w)).share("tempram");
+	map(0xe0, 0xff).ram().w(FUNC(ht1130_device::displayram_w)).share("displayram");
+}
+
+void ht1190_device::internal_data_map_ht1190(address_map &map)
+{
+	map(0x00, 0x9f).ram().w(FUNC(ht1190_device::tempram_w)).share("tempram");
+	map(0xb0, 0xff).ram().w(FUNC(ht1190_device::displayram_w)).share("displayram");
+}
+
+void ht1130_device::device_start()
+{
+	space(AS_PROGRAM).cache(m_cache);
+	space(AS_PROGRAM).specific(m_space);
```

Comment:
Why do you have both a `cache` and a `specific` here?  You should choose the one that’s likely to perform best and use it.